content/livesync/frontend.textile

```diff
@@ -0,0 +1,373 @@
+```
+
+Import the SDKs into your project:
+
+```[javascript]
+import ModelsClient from '@ably-labs/models';
+import { Realtime } from 'ably';
+```
+
+Instantiate a realtime client using the Ably JavaScript SDK and pass the generated client into the Models constructor:
+
+```[javascript]
+const ably = new Realtime({ key: '...' });
+const modelsClient = new ModelsClient({ ably });
+```
+
+h3(#client-options). ClientOptions
+
+<aside data-type='note'>
+<p>View the "API references":https://sdk.ably.com/builds/ably-labs/models/main/typedoc/ for the Models SDK.</p>
+</aside>
+
+In addition to the underlying Ably realtime client, you can provide a number of other "@ClientOptions@":/api/realtime-sdk/types#client-options to configure the behavior of the Models SDK:
+
+- @syncOptions@ := is used to configure how the model state is synchronised via the "sync function":#sync.
+- @historyPageSize@ := is the limit used when "querying for paginated history":/livesync/models/models#history used to subscribe to changes from the correct point in the channel.
+- @messageRetentionPeriod@ := is the message retention period configured on the channel. This is used to determine whether the model state can be brought up to date from "message history":/livesync/models/models#history rather than via a re-sync.
+- @retryStrategy@ := defines a retry strategy to use if calling the "sync function":/livesync/models/models#sync throws an error.
+- @eventBufferOptions@ := used to configure the in-memory sliding-window buffer used for reordering and deduplication.
+- @optimisticEventOptions@ := is used to configure how "optimistic events":/livesync/models/models#optimistic are applied.
+- @logLevel@ := configures the log level used to control the verbosity of log output. One of @fatal@, @error@, @warn@, @info@, @debug@, or @trace@.
+- @ModelsClient@ := captures a collection of named model instances used in your application and provides methods for creating new models.
+
+The following is an example of setting @ClientOptions@ when instantiating the Models SDK:
+
+```[javascript]
+const modelsClient = new ModelsClient({
+ client,
+ logLevel,
+ syncOptions: {
+  historyPageSize,
+  messageRetentionPeriod,
+  retryStrategy,
+ },
+ eventBufferOptions: {
+  bufferMs,
+  eventOrderer,
+ },
+ optimisticEventOptions: {
+  timeout,
+ },
+});
+```
+
+h2(#create-or-retrieve). Create or retrieve a model
+
+Create a model using the @models.get()@ method on the client. If you specify a name that already exists, it will be returned.
+
+Instantiate a model using a unique channel name. A unique channel name identifies the model on the client and corresponds to the "channel":/channels name used to subscribe to state updates from the backend:
+
+```[javascript]
+const model = modelsClient.models.get({
+  name: 'Posts',
+  channelName: 'posts',
+  sync,
+  merge,
+});
+```
+
+TODO: Diagram of how the model components work together
+
+h2(#designing). Designing your Channels and Models
```

Comment:
I think this is an interesting question. It could go one of two ways. If you're trying to get set up or a basic understanding of LiveSync and how to implement it or just getting started do you want to know this first? Or is this a piece of information you'd want to learn as you build your actual intended application? 

This was discussed quite a bit on Wednesday. We have too much information overload on most of our docs pages that may never be needed for the developer, or will only be needed later. May or may not be for this subject. But yeah. It could go either side I think.